driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -86,10 +89,13 @@
 
   private AtomicInteger[] scheduledForCreation;
 
-  private final EventExecutor timeoutsExecutor;
+  private final EventExecutor scheduler;
 
   private final AtomicReference<CloseFuture> closeFuture = new AtomicReference<CloseFuture>();
 
+  private final BlockingQueue<ConnectionTask> pendingTasks =
```

Comment:
Please don't mix two unrelated changes in one commit. Rate limiting should be one commit and using local port should be another.